### ISSUE TITLE
Support label formatters

### ISSUE THREAD

**Issue by slavafomin** [isteven/angular-multi-select#243]
Allow function to be passed to the `button-label-formatter` and `item-label-formatter` attributes.

When component will need to generate the label for button or item, it will call this function (we can call such a function «label formatter») and pass actual item element[s] to it.

# Example

```javascript
function controller ($scope) {
  $scope.elementList = [
    { firstName: 'Slava', lastName: 'Fomin', email: 's.fomin@betsol.ru' }
    // ...
  ];
  $scope.itemLabelFormatter = function (item) {
    return (item.firstName + ' ' + item.lastName + ' <' + item.email + '>');
  }
}
```

```html
<isteven-multi-select
    input-model="elementList"
    item-label-formatter="itemLabelFormatter"
>
</isteven-multi-select>
```

It will allow developers to dynamically format labels as they see fit.

**Comment by isteven** [comment]
Hi @slavafomin , 

Thanks for the idea. Actually my goal is to allow user to inject their own HTML template (a la $templateCache thing), but we'll see. For now seems that most of the users are happy with what's available. .

**Comment by slavafomin** [comment]
You should consider that most of the people will never tell or even understand what they really need. Some people will just skip the library if they will not like it's implementation.

The template approach looks promising however.

**Comment by isteven** [comment]
I totally respect their decision to use, modify, or skip my library :)

**Comment by tedyoung** [comment]
+1 on injecting HTML or some other mechanism to control the formatting (I was making similar modifications in my fork).

**Comment by tufan-yoc** [comment]
+1 for label formatters. Current pre generated labels simply don't look good on some designs. 

> I totally respect their decision to use, modify, or skip my library :)
telling users to go to "somewhere else", is not "nice", also not the right approach for a product development.

**Comment by aljamala** [comment]
I agree,  I've been integrating your directive, but after reading some of your comments regarding proposed updates (which would definitely make it more robust), I am reconsidering my choice.  It seems I can't even add brackets for my labels? Which is a common way to represent abbreviations.

**Comment by isteven** [comment]
Hi @aljamala ,

I did put brackets in some of my earlier examples. No problem with them.

**Comment by aljamala** [comment]
It's not the data that contains brackets.  For example I want to be able to do the following: item-label="name (abbreviation)", and I don't see that output unless I remove the brackets.

**Comment by isteven** [comment]
@aljamala ,

Oh, that.. You are correct, at the moment they are not supported.

**Comment by RyanPicanco** [comment]
I've forked this and made the button and itemLabels be a template type string. Feel free to use that solution if you like.

**Comment by chris-wood-dynmark** [comment]
+1 for label formatting, I am having to consider using another directive purely because I cannot display the items on two lines in the select list e.g
Name
Description

**Comment by simonered** [comment]
+1 for label formatting

**Comment by jrahulroy** [comment]
+1

**Comment by rbordeanu** [comment]
+1

**Comment by mitomm** [comment]
+1

**Comment by miriame** [comment]
+
